content/en/blog/2019/knative-activator-adapter/index.md

```diff
@@ -0,0 +1,94 @@
+---
+title: Mixer out-of-process adapter for Knative
+subtitle: Demonstrates a Mixer out-of-process adapter which implements the Knative scale-from-zero logic
+description: Demonstrates a Mixer out-of-process adapter which implements the Knative scale-from-zero logic.
+publishdate: 2019-08-27
+attribution: Idan Zach (IBM)
+keywords: [mixer,adapter,knative,scale-from-zero]
+---
+
+This post demonstrates how you can use [Mixer](https://istio.io/faq/mixer/) to push application logic
+into Istio. It describes a Mixer adapter which implements the [Knative](https://knative.dev/) scale-from-zero logic
+with simple code and similar performance to the original implementation.
+
+## Knative serving
+
+[Knative Serving](https://knative.dev/docs/serving/) builds on [Kubernetes](https://kubernetes.io/) to support deploying
+and serving of serverless applications. A core capability of serverless platforms is scale-to-zero
+functionality which reduces resource usage and cost of inactive workloads.
+A new mechanism is required to scale from zero when an idle application receives a new request.
+
+The following diagram represents the current Knative architecture for scale-from-zero.
+
+{{< image width="60%" link="knative-activator.png" caption="Knative scale-from-zero" >}}
+
+The traffic for an idle application is redirected to **Activator** component by programming Istio with `VirtualServices`
+and `DestinationRules`. When **Activator** receives a new request, it:
+
+1. buffers incoming requests
+1. triggers the **Autoscaler**
+1. redirects requests to the application after it has been scaled up, including retries and load-balancing (if needed)
+
+Once the application is up and running again, Knative restores the routing from **Activator** to the running application.
+
+## Mixer adapter
+
+[Mixer](https://istio.io/faq/mixer/) provides a rich intermediation layer between the Istio components and infrastructure backends.
+It is designed as a stand-alone component, separate from [Envoy](https://www.envoyproxy.io/), and has a simple extensibility model
+to enable Istio to interoperate with a wide breadth of backends. Mixer is inherently easier to extend
+than Envoy is.
+
+Mixer is an attribute processing engine that uses operator-supplied configuration to map request attributes from the Istio proxy into calls
+to the infrastructure backends systems via a pluggable set of adapters. Adapters enable **Mixer** to expose a single consistent API, independent of the
+infrastructure backends in use. The exact set of adapters used at runtime is determined through operator configuration and can easily
+be extended to target new or custom infrastructure backends.
+
+Here you utilized these benefits to implement the Knative scale-from-zero logic in a special Mixer
```

Comment:
curious about the "Here you..." bit. Seems a bit out of the flow of the rest of the document. I might also be tempted to avoid talking about prior releases and adapter models.

Maybe something like: 

```text
In order to achieve Knative scale-from-zero, we use a Mixer out-of-process adapter to call the **Autoscaler**. Out-of-process adapters for Mixer allow developers to... This enables...

This new out-of-process adapter will be called as part of the authorization request originally sent to Mixer. The following diagram...
```